src/main/kotlin/io/github/zap/build/gradle/convention/ProjectExtension.kt

```diff
@@ -169,4 +171,99 @@ fun Project.configureShadow(destination: File,
     tasks.named("build", DefaultTask::class.java) {
         it.dependsOn(sj)
     }
+}
+
+fun Project.qs (name: String = "", options: ExternalModuleDependency.() -> Unit = {}) : ExternalModuleDependency.() -> Unit {
+    return {
+        val actualName = name.ifEmpty { this.name }
+        version {
+            if (isLocalTag(actualName)) {
+                it.require(getLocalVersion(actualName))
+            } else {
+                if(version != null && version!!.isNotEmpty()) {
+                    ext["qs-$actualName-ver"] = version
+                } else {
+                    val ver = getVerFromAncestor(actualName)
+                    if (ver != null) {
+                        it.require(ver)
+                    } else {
+                        logger.log(LogLevel.ERROR, "No version specified for dependency $actualName, fallback to 0.0.0-SNAPSHOT")
+                        it.require("0.0.0-SNAPSHOT")
+                    }
+                }
+            }
+        }
+        options()
+    }
+}
+
+@Suppress("UNCHECKED_CAST")
+internal fun Project.getQsLocals(): Set<String>? {
+    if(ext.has("qs-useLocals")) {
+        val cached = ext["qs-useLocals"]
+        if(cached is Set<*>) {
+            return cached as Set<String>
+        }
+    }
+
+    val useLocalClause = project.findProperty("qs.useLocal") as String?
+    if(useLocalClause == null) {
+        ext["qs-useLocals"] = null
+        return null
+    }
+
+    val useLocals = useLocalClause.split(",")
+        .map { it.trim() }
+        .filter { it.trim().isNotEmpty() }
+        .toSet()
+        ext["qs-useLocals"] = useLocals
+        return useLocals
+    }
+
+internal fun Project.isLocalTag(name: String): Boolean {
+    val locals = getQsLocals() ?: return false
+    if(locals.isEmpty()) return true
+
+    return locals.contains(name)
```

Comment:
in operator cooler 👿
this can also be one line in combination with the last line